assets/model_monitoring/components/src/action_analyzer/action_analyzer_correlation_test/run.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Action Analyzer correlation test."""
+
+import argparse
+from scipy import stats
+from pyspark.sql.types import (
+    StructType,
+    StructField,
+    StringType,
+    FloatType
+)
+from shared_utilities.io_utils import (
+    try_read_mltable_in_spark,
+    save_spark_df_as_mltable,
+    create_spark_df,
+    save_empty_dataframe
+)
+
+
+def get_output_schema() -> StructType:
+    """Get Action Data Spark DataFrame Schema."""
+    schema = StructType(
+        [
+            StructField("index_id", StringType(), True),
+            StructField("group", StringType(), True),
+            StructField("confidence_score", FloatType(), True)
+        ]
+    )
+    return schema
+
+
+def save_action_data(action_rows, output_path):
+    """Save Action Data Spark DataFrame."""
+    schema = get_output_schema()
+    df = create_spark_df(action_rows, schema)
+    save_spark_df_as_mltable(df, output_path)
+
+
+def generate_action_rows(pdf, index_set, group_set):
+    """Generate the action data."""
+    actions_row = []
+    for index in index_set:
+        for group in group_set:
+            print("Group: ", group)
+            metrics = group.split("_")[0]
+            good_group_name = f"{metrics}_good_group"
+            if group == good_group_name:
+                print(f"Skip {good_group_name}, only do t-test for bad group")
+                continue
+            index_df = pdf[pdf['index_id'] == index]
```

Comment:
index_id, index_score,group_list, those hard coded const value, make sure we don't have those columns

Reply:
Make them as constants. Also in the first step, only taking the columns from GSQ, which will not have extra fields like these